trainee_template/react_series.amlg

```diff
@@ -678,25 +718,44 @@
 
 			;return series_data so it's stored as previous_result for accumulation in the next iteration of the while loop
 			;and explanation details if requested
-			(if (or output_influential_cases output_cap)
+			(if output_details
 				(seq
-					(call !NormalizeReactSeriesInfluentialCases)
-
-					;update all_influential_cases on the last iteration so it can be output when react_series returns
-					(if (and done (size influential_cases))
-						;wrap the influential cases in a list so it's accumulated as a list of lists of assocs (a list per row)
-						(accum (assoc all_influential_cases (list influential_cases) ))
+					(if (contains_index case_detail_values_map "influential_cases")
+						(call !NormalizeReactSeriesInfluentialCases)
 					)
 
-					(if (and done (size local_class_probabilities_map))
-						(accum (assoc all_categorical_action_probabilities (list local_class_probabilities_map)))
+					;update series_detail_values_map on the last iteration so it can be output when react_series returns
+					(if done
+						(assign (assoc
+							series_detail_values_map
+								(map
+									(lambda
+										(if (size (get case_detail_values_map (current_index)))
+											(append (current_value) (list (get case_detail_values_map (current_index 1))) )
+
+											;don't append if the case detail value is empty
+											(current_value)
+										)
+									)
+									series_detail_values_map
+								)
+						))
 					)
 
-					;store into previous_result as a tuple of [series_data, all_categorical_action_probabilities, all_influential_cases]
-					(list
-						series_data
-						(append all_categorical_action_probabilities (list local_class_probabilities_map))
-						(append all_influential_cases (list influential_cases))
+					;store accumulated data into previous_result as an assoc
+					(append
```

Comment:
this makes it append assoc for every row regardless whether there are details. in most flows, there aren't details and this is unnecessary, i'd refactor this to be inside the `if output_details` :
`(if output_details
    (append ... )
    ;else just output the series data
    (assoc "series_data" series_data)
)`

Reply:
This is actually already inside of a `(if output_details`.